aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1119,6 +1139,54 @@ public <T extends S3ClientBuilder> void applyS3ServiceConfigurations(T builder)
                 .build());
   }
 
+  /**
+   * Configure a signer for an S3 client.
+   *
+   * <p>Sample usage:
+   *
+   * <pre>
+   *     S3Client.builder().applyMutation(awsProperties::applyS3SignerConfiguration)
+   * </pre>
+   */
+  public <T extends S3ClientBuilder> void applyS3SignerConfiguration(T builder) {
+    if (null != s3SignerImpl) {
+      builder.overrideConfiguration(
+          c -> c.putAdvancedOption(SdkAdvancedClientOption.SIGNER, loadS3SignerDynamically()));
+    }
+  }
+
+  @SuppressWarnings("checkstyle:NestedTryDepth")
+  private Signer loadS3SignerDynamically() {
+    // load the signer implementation dynamically
+    Object signer;
+    try {
+      signer =
+          DynMethods.builder("of")
```

Comment:
This seems suspicious to me. We don't use a static `of(Map)` for any other dynamically loaded class.

Reply:
This is the method from the `S3V4RestSignerClient`:
```
public static S3V4RestSignerClient of(Map<String, String> properties) {
    return ImmutableS3V4RestSignerClient.builder().properties(properties).build();
  }
```
We could also rename this if required

Reply:
I think we should follow the standard conventions for dynamic loading. This should always call the no-arg constructor and then call `initialize` with the properties map. If we don't have an interface with the `initialize` method, then it can be called dynamically as well.

Reply:
I've tried to outline in https://github.com/apache/iceberg/pull/6169#discussion_r1031555625 why I'd like to keep this an Immutable class. Given that an Immutable class doesn't provide a no-arg constructor (at least not out-of-the-box), we can't use our conventional way of dynamically loading this class (no-arg constructor + initialize method) unfortunately

Reply:
I think I get why we want to break the pattern of dynamic initialization because of using Immutable, I will leave that part to Ryan to make a call to use that for the signer class or not. But can we at least use a method name that is more intuitive to readers of this code, such as `initialize`, instead of `of`?

Reply:
Looks like we could use the style annotations to change it to something else ([they recommend](https://immutables.github.io/style.html#Apply_Style) `new`):

```java
@Value.Style(builder = "new")
```

Reply:
Actually, based on the docs, I think that switches the instantiation to a constructor method, so it might be possible to get closer to the default behavior if we can construct with parameters.

Reply:
we discussed this offline with @danielcweeks and decided to name the method `create`. 
/cc @jackye1995